### PR TITLE
FT: Node v4.5.0 and new buffer api switch

### PR DESCRIPTION
**Pull request by alexandre-merle**
Update to the new buffer api

### CODE REVIEW

File: tests/unit/api/bucketDelete.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const canonicalID = 'accessKey1';
 const authInfo = makeAuthInfo(canonicalID);
 const namespace = 'default';
 const bucketName = 'bucketname';
-const postBody = new Buffer('I am a body');
+const postBody = Buffer.from('I am a body', 'binary');
```
Comment (rahulreddy) [inline tests/unit/api/bucketDelete.js]: This should be `utf8` as it identifies character encoding of the text provided.
```js
> Buffer.from('pâtisserie', 'binary').toString()
'p�tisserie'
> Buffer.from('pâtisserie').toString()
'pâtisserie'
```

File: tests/unit/api/bucketGet.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ const bucketName = 'bucketname';
 const delimiter = '/';
 const log = new DummyRequestLogger();
 const namespace = 'default';
-const postBody = new Buffer('I am a body');
+const postBody = Buffer.from('I am a body', 'binary');
```
Comment (rahulreddy) [inline tests/unit/api/bucketGet.js]: Same here. It should be utf8 encoding.

File: tests/unit/api/deletedFlagBucket.js [hunk 3]
```diff
@@ -224,7 +224,7 @@ describe('deleted flag bucket handling', () => {
             const setUpRequest = createAlteredRequest({}, 'headers',
             baseTestRequest, baseTestRequest.headers);
             setUpRequest.objectKey = objName;
-            const postBody = new Buffer('I am a body');
+            const postBody = Buffer.from('I am a body', 'binary');
```
Comment (rahulreddy) [inline tests/unit/api/deletedFlagBucket.js]: It should be utf8 encoding.

File: tests/unit/api/deletedFlagBucket.js [hunk 4]
```diff
@@ -251,7 +251,7 @@ describe('deleted flag bucket handling', () => {
         const setUpRequest = createAlteredRequest({}, 'headers',
         baseTestRequest, baseTestRequest.headers);
         setUpRequest.objectKey = 'objectName';
-        const postBody = new Buffer('I am a body');
+        const postBody = Buffer.from('I am a body', 'binary');
```
Comment (rahulreddy) [inline tests/unit/api/deletedFlagBucket.js]: It should be utf8 encoding.

File: tests/unit/api/multipartUpload.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ const authInfo = makeAuthInfo(canonicalID);
 const namespace = 'default';
 const bucketName = 'bucketname';
 const mpuBucket = `${constants.mpuBucketPrefix}${bucketName}`;
-const postBody = new Buffer('I am a body');
+const postBody = Buffer.from('I am a body', 'binary');
```
Comment (rahulreddy) [inline tests/unit/api/multipartUpload.js]: It should be utf8 encoding.

File: tests/unit/api/multipartUpload.js [hunk 6]
```diff
@@ -89,7 +89,7 @@ describe('Multipart Upload API', () => {
             // until here
             const testUploadId = json.InitiateMultipartUploadResult.UploadId[0];
             const md5Hash = crypto.createHash('md5');
-            const bufferBody = new Buffer(postBody, 'binary');
+            const bufferBody = Buffer.from(postBody, 'binary');
```
Comment (rahulreddy) [inline tests/unit/api/multipartUpload.js]: encoding is not a valid argument here as the input itself is a buffer. Here's the example from node's docs
```js
const buf1 = Buffer.from('buffer');
const buf2 = Buffer.from(buf1);
```

Comment (alexandre-merle) [inline tests/unit/api/multipartUpload.js]: right :)

File: tests/unit/api/bucketDelete.js (hunk 1 above)
Comment (rahulreddy) [inline tests/unit/api/bucketDelete.js]: No, it's actually utf8 only. I did the same mistake when I did the binary buffers effort in metadata and bucketclient.

Comment (alexandre-merle) [inline tests/unit/api/bucketDelete.js]: hmm, that means only cryptographic encoding is binary by default, will change that :)

**Comment by michael-zapata** [comment]
I'd rather have the encoding argument explicit, helps a lot with readability.

**Comment by alexandre-merle** [comment]
Updated

**Comment by alexandre-merle** [comment]
@ironman-machine let's do it

File: lib/kms/common.js [hunk 7]
```diff
@@ -78,7 +78,7 @@ export default class Common {
              * bumping the cryptoScheme number saved in the object
              * metadata along with the dataKey.
              */
-            const salt = new Buffer('ItsTasty');
+            const salt = Buffer.from('ItsTasty', 'binary');
```
Comment (LaurenSpiegel) [inline lib/kms/common.js]: There are still some 'binary' references here.

Comment (alexandre-merle) [inline lib/kms/common.js]: right :) will update it

**Comment by alexandre-merle** [comment]
Updated :)

**Comment by alexandre-merle** [comment]
@ironman-machine i'm sure you're ok now :)

SCALITY_INTEGRATION_BRANCH=ft/update-node-minor

**Comment by alexandre-merle** [comment]
@LaurenSpiegel end to end passed :)

**Pull request merged**